app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -1285,56 +1252,46 @@ export default Controller.extend(bufferedProperty("model"), {
   },
 
   _modifyTopicBookmark(bookmark) {
-    return openBookmarkModal(
-      bookmark,
-      {
-        onAfterSave: (savedData) => {
-          this._syncBookmarks(savedData);
-          this.model.set("bookmarking", false);
-          this.model.set("bookmarked", true);
-          this.model.incrementProperty("bookmarksWereChanged");
-          this.appEvents.trigger(
-            "bookmarks:changed",
-            savedData,
-            bookmark.attachedTo()
-          );
+    return openBookmarkModal(bookmark, {
+      onAfterSave: (savedData) => {
+        this._syncBookmarks(savedData);
+        this.model.set("bookmarking", false);
+        this.model.set("bookmarked", true);
+        this.model.incrementProperty("bookmarksWereChanged");
+        this.appEvents.trigger(
+          "bookmarks:changed",
+          savedData,
+          bookmark.attachedTo()
+        );
 
-          // TODO (martin) (2022-02-01) Remove these old bookmark events, replaced by bookmarks:changed.
-          this.appEvents.trigger("topic:bookmark-toggled");
-        },
-        onAfterDelete: (topicBookmarked, bookmarkId) => {
-          this.model.removeBookmark(bookmarkId);
-        },
+        // TODO (martin) (2022-02-01) Remove these old bookmark events, replaced by bookmarks:changed.
```

Comment:
Is this the time when the TODO was added or its deadline?

Reply:
Hmm that's the "deadline" (self imposed by me :laughing: ) , still need to remove this, will do so in another PR.